internal/executor/utilisation/cluster_utilisation.go

```diff
@@ -165,24 +172,63 @@ func (clusterUtilisationService *ClusterUtilisationService) GetAllNodeGroupAlloc
 		return []*NodeGroupAllocationInfo{}, err
 	}
 
+	batchPods, err := clusterUtilisationService.clusterContext.GetBatchPods()
+	if err != nil {
+		// Failing this step will just mean cordoned utilisation won't be accounted for
+		// This isn't a critical failure so we log and continue
+		log.Errorf("Error fetching batch pods to calculate cordond node utilisation: %v", err)
+	}
+
 	nodeGroups := clusterUtilisationService.nodeInfoService.GroupNodesByType(allAvailableProcessingNodes)
 	result := make([]*NodeGroupAllocationInfo, 0, len(nodeGroups))
 
 	for _, nodeGroup := range nodeGroups {
 		totalNodeResource := common.CalculateTotalResource(nodeGroup.Nodes)
 		allocatableNodeResource := allocatableResourceByNodeType[nodeGroup.NodeType.Id]
+		cordonedNodeResource := clusterUtilisationService.getCordonedResource(nodeGroup.Nodes, batchPods)
 
 		result = append(result, &NodeGroupAllocationInfo{
 			NodeType:                     nodeGroup.NodeType,
 			Nodes:                        nodeGroup.Nodes,
 			NodeGroupCapacity:            totalNodeResource,
 			NodeGroupAllocatableCapacity: allocatableNodeResource,
+			NodeGroupCordonedCapacity:    cordonedNodeResource,
 		})
 	}
 
 	return result, nil
 }
 
+// getCordonedResource takes a list of nodes and a list of pods and returns the resources allocated
+// to pods running on cordoned nodes. We need this information in calculating queue fair shares when
+// significant resource is running on cordoned nodes.
+func (clusterUtilisationService *ClusterUtilisationService) getCordonedResource(nodes []*v1.Node, pods []*v1.Pod) common.ComputeResources {
+	cordonedNodes := util.FilterNodes(nodes, func(node *v1.Node) bool { return node.Spec.Unschedulable })
+	podsOnNodes := GetPodsOnNodes(pods, cordonedNodes)
+	usage := common.ComputeResources{}
+	for _, pod := range podsOnNodes {
+		for _, container := range pod.Spec.Containers {
+			containerResource := castAsComputeResources(container.Resources.Limits) // Not 100% on whether this should be Requests or Limits
+			usage.Add(containerResource)
+		}
+	}
+	fmt.Printf("%v\n", usage)
+	return usage
+}
+
+// castAsComputeResources casts a ResourceList type as common.ComputeResources.
+// Both are aliases for map[string]resource.Quantity.
+// This feels dirty, there must be a way to directly cast them (remove this line or replace this function)
```

Comment:
Perhaps just remove the one in common? If the right abstraction exists in the Kubernetes API, we may not need to create our own version. I suggest adding TODO to the start of the sentence and putting a suggested fix into the comment.

Reply:
As far as I can tell you, the reason we do this is because in go you can't extend structs outside the package they're defined in.

We have a good number of methods we put on top of common.ComputeResources, for instance termwise, mutative addition and subtraction (https://github.com/G-Research/armada/blob/master/internal/common/resource.go).

Maybe the question we should be asking is why we don't make these normal functions (maybe make them pure functions rather than modifying the subject as well) on map[string]resource.Quantity? and then we *should* be able to cast that directly to v1.ResourceList. That would be cleaner to my understanding of golang wanting methods only to be for implementing interfaces (which we don't actually do in the case of common.ComputeResources).

Reply:
So I think this is just `common.ComputeResources.FromResourceList` isn't it?  You can probably delete this function and just use that.

As an aside, what you're doing here isn't really a cast, but rather a conversion.